README.md

```diff
@@ -2,6 +2,55 @@
 `dist_object` implementation using HPX component
 
 # Distributed Objects
+Distributed object in HPX: a single logical object partitioned over a set of localities/nodes/machines, where every locality shares the same global name for the distributed object (i.e. a universal name), but owns its local value. In other words, local data of the distributed object can be different but share the same globally.
+```cpp
+dist_object<double> dist_double("a_unique_string", 42.0);
+```
+Each locality in a given collection of the localities must call a constructor collectively for `dist_object<T>`, with a unique user-provided string name and a value of type T representing the locality’s instance value for the distributed object. The string name is required for the purpose of enabling HPX globally registers and retrives the distributed object.
+
+The `run_accumulation_reduce_to_locality0_parallel` code below demonstrates a simple accumulation computation using the distributed object in HPX.
+
+In this example, the distributed object is an integer over all localities, the value of the object in each locality is set to the index value of the locality, such that (`dist_int` in locality 0 is 0, `dist_int` in locality 1 is 1, etc.) Although the constructor for a distributed object is called collectively, it is not cleaer that when the constructor on each given locality will complete its own construction. If we make a fetch call to the distributed object in a remote locality, which has not been constructed or ready, it will cause hazards. Thus, it is necessary to insert a barrier to ensure that the object is being constructed on all localities and they are on the same page. A barrier is used to guarantee rank 0 will not access any remote instance of the distributed object until all the ranks have provided their construction. To notice,  the `fetch` is done through an asychronously fashion, and returns a future of a copy of the instance of this distributed object associated with the given locality. The test of accumulation is set and varified on locality 0.
```

Comment:
- [ ] "it is not cleaer" spelling. 

- [ ] Wording on the second half of that sentence is also not clear at all. 

- [ ] "it will cause errors" or "represents a hazard", not "it will cause hazards". 

- [ ] "on the same page" is informal English, and shouldn't be there (at the very least from a style perspective). 

- [ ] Change "a barrier is used to guarantee rank 0 will not access any remote instance" to "A barrier is used to guarantee no rank will access any remote instance".

- [ ] The second to last sentence should be revised as well to something more like "Importantly, the 'fetch' is done asynchronously, and returns a future representing a copy of the data which the given locality's distributed object owns."

- [ ] I'm not sure what you mean by "is set" in the last sentence. 

- [ ] Spelling on "varified"